content/courses/Ada_For_The_C_Embedded_Developer/chapters/07_Reusability.rst

```diff
@@ -766,7 +766,150 @@ callback routine.
 Design by components using dynamic libraries
 --------------------------------------------
 
-.. todo::
+In the previous sections, we have shown how to use packages to create separate
+components of a system. As we know, when designing a complex system, it is
+advisable to separate concerns into distinct units, so we can use Ada packages
+to represent each unit of a system. In this section, we go one step further and
+create separate dynamic libraries for each component, which we'll then link to
+the main application.
 
```

Comment:
I wonder if we should mention the example that comes with GNAT that does this (there's a paper too). it is in the GNAT examples dir, 

share\examples\gnat\plugins

Reply:
I've look into that example, but it seems a little bit complicated for an introductory course like this one. In my opinion, we should instead address the topic in one of the _Embedded Ada_ courses.

Reply:
I agree that it is too much for the course to explore it. I'm only wondering if a reference to it would be useful, in case they want to see one they can play with.

Reply:
Do you know any online documentation I could link to? Otherwise, I'd just include a todo item here, so that we don't forget to link to the _Embedded Ada_ course when it's available.

Reply:
None that I know of. I was thinking only of mentioning that it is in the examples included with the compiler, in the "plugin" directory.
(But in any case I don't see this as a section for the Embedded course, to be honest.)

Reply:
Sorry, I misunderstood you: I thought you wanted me to cover this more advanced topic here... Just mentioning the GNAT example is perfectly fine, of course, so I've just replaced the todo item by an admonition.